frontend/apps/reader/modules/readerpagemap.lua

```diff
@@ -327,127 +357,207 @@ function ReaderPageMap:onDocumentRerendered()
+                        callback = function(touchmenu_instance)
+                            UIManager:show(SpinWidget:new{
+                                title_text = _("Page labels font size"),
+                                value = self.label_font_size,
+                                value_min = 8,
+                                value_max = 20,
+                                default_value = self.label_default_font_size,
+                                keep_shown_on_apply = true,
+                                callback = function(spin)
+                                    self.label_font_size = spin.value
+                                    G_reader_settings:saveSetting("pagemap_label_font_size", self.label_font_size)
+                                    touchmenu_instance:updateItems()
+                                    self:resetLayout()
+                                    self:updateVisibleLabels()
+                                    UIManager:setDirty(self.view.dialog, "partial")
+                                end,
+                            })
+                        end,
+                    },
+                },
+                separator = true,
+            },
+            -- current book
+            {
+                text_func = function()
+                    return T(_("Characters per synthetic page: %1"), self.chars_per_synthetic_page or _("disabled"))
                 end,
                 keep_menu_open = true,
+                callback = function()
+                    UIManager:show(SpinWidget:new{
+                        title_text =  _("Characters per page"),
+                        value = self.chars_per_synthetic_page or self.chars_per_synthetic_page_default,
+                        value_min = 512,
+                        value_max = 4096,
+                        default_value = self.chars_per_synthetic_page_default,
+                        ok_always_enabled = true,
+                        keep_shown_on_apply = true,
+                        callback = function(spin)
+                            self:applyCharsPerSyntheticPage(spin, spin.value)
+                        end,
+                        extra_text = _("Disable"),
+                        extra_callback = function(spin)
+                            self:applyCharsPerSyntheticPage(spin)
+                        end,
+                    })
+                end,
             },
             {
-                text = _("Reference page numbers list"),
+                -- @translators This shows the <dc:source> in the EPUB that usually tells which hardcopy edition the reference page numbers refers to.
+                text_func = function()
+                    if self.has_pagemap then
+                        if self.chars_per_synthetic_page then
+                            return _("Reference source info: synthetic")
+                        end
+                        return _("Reference source info: built-in")
```

Comment:
May be:
if self.ui.document:getPageMapSource() then "Reference source info: publisher"
else "Reference source info: none publisher provided" (or something)
(dunno if it returns nil or `""` when empty)